### ISSUE TITLE
Translations in v2.2 seem broken

### ISSUE THREAD

**Issue by cicnavi** [simplesamlphp/simplesamlphp#2009]
In admin area of SimpleSAMLphp v2.2.0 instance, if I change language using the language selector - nothing happens, that is, only 'msgid' is shown in UI for each string that should be translated.

It think it has something to do with domains. When parsing po files, translations are groped by domains (in contrast to mo files, when that was not the case). If I hardcode domain to null in vendor/gettext/translator/src/Generator/ArrayGenerator.php::generateArray() (excerpt below), then it works. 

```php
        return [
            //'domain' => $translations->getDomain(),
            'domain' => null,
            'plural-forms' => $translations->getHeaders()->get(Headers::HEADER_PLURAL),
            'messages' => $messages,
        ];
```

**Comment by tvdijen** [comment]
Confirmed!  I'm not sure what you mean with the domains though.. We group translations by domain (= module name) and I can still see the domains being loaded in the debug logs.

**Comment by cicnavi** [comment]
When it works, If I dump property 'dictionary' from vendor/gettext/translator/src/Translator.php, it looks like this:

```
array (size=1)
  '' => 
    array (size=1)
      '' => 
        array (size=233)
          'A service has requested you to authenticate yourself. Please enter your username and password in the form below.' => string 'Un servicio solicita que se autentique. Esto significa que debe indicar su nombre de usuario y su clave de acceso en el siguiente formulario.' (length=141)
          'ADFS IdP Metadata' => string 'Metadatos IdP ADFS' (length=18)
...
```

When it doesn't work, it looks like this (note the 'messages' key / domain):

```
array (size=1)
  'messages' => 
    array (size=1)
      '' => 
        array (size=233)
          'A service has requested you to authenticate yourself. Please enter your username and password in the form below.' => string 'Un servicio solicita que se autentique. Esto significa que debe indicar su nombre de usuario y su clave de acceso en el siguiente formulario.' (length=141)
          'ADFS IdP Metadata' => string 'Metadatos IdP ADFS' (length=18)
...
```

**Comment by tvdijen** [comment]
You mean 'when it works' if you hardcode the domain to `null`?
`messages` is the default domain.. In case of the admin module, I would expect the key to be `admin`.

**Comment by tvdijen** [comment]
@monkeyiq I don't understand what you did to revert the translations-patch.. I still see .mo file related code and I still see scripts in the composer.json file..

**Comment by monkeyiq** [comment]
The scripts that were making the mo files seem to be still removed in 2.2? https://github.com/simplesamlphp/simplesamlphp/blob/8123037afc45b1b01fcda83e0c52cf0d54d8b787/composer.json#L126

**Comment by monkeyiq** [comment]
Interesting. I had some mo files in my sspdev web server directory left over :/ 

Anyway, once I remove them setting the language to French in the admin does not work as expected. The hiccup is that when loading from the po files the domain from there is preserved. This was placing all the French admin strings into an admin domain in a `var_dump($this->translator)` and I don't think they were bring found there. If, on the other hand you wipe the domain after reading a po file using the following then the language changes as one would expect.
```
$translations->setDomain('');
```

**Comment by cicnavi** [comment]
> You mean 'when it works' if you hardcode the domain to `null`?
> `messages` is the default domain.. In case of the admin module, I would expect the key to be `admin`.

Yes, exactly. There is also admin key, I just didn't paste it. It comes afterward...

**Comment by tvdijen** [comment]
I think this has to do with the fact that po-files are now automatically updated/regenerated..
In the 2.0-branch the .po files didn't have the `X-domain` header.

**Comment by monkeyiq** [comment]
Looking at a slightly older release, simplesamlphp-2.1.1-full, the admin.po file has an `X-Domain: admin` but it seems that `msgunfmt admin.mo` does not show a domain. 

If I copy admin.po to test.po and msgfmt that test.po to get messages.mo and msgunfmt that messages.mo then I see a domain. So the gettext commands are happy to preserve the domain.

So it seems maybe that the processing of the admin.po file into admin.mo in ssp might have been removing the x-domain. Other code that then used the translations were relying on the domain not being set in the binary.

**Comment by tvdijen** [comment]
I think you're right.. I'm just a little confused what happened to our (module) domains over time.. Apparently everything is loaded into the default domain.

**Comment by tvdijen** [comment]
Fixed & tagged 2.2.1
